extensions/cloudfiles-extensions/pom.xml

```diff
@@ -0,0 +1,195 @@
+<?xml version="1.0"?>
+<!-- ~ Druid - a distributed column store. ~ Copyright 2012 - 2015 Metamarkets 
+	Group Inc. ~ ~ Licensed under the Apache License, Version 2.0 (the "License"); 
+	~ you may not use this file except in compliance with the License. ~ You 
+	may obtain a copy of the License at ~ ~ http://www.apache.org/licenses/LICENSE-2.0 
+	~ ~ Unless required by applicable law or agreed to in writing, software ~ 
+	distributed under the License is distributed on an "AS IS" BASIS, ~ WITHOUT 
+	WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. ~ See the 
+	License for the specific language governing permissions and ~ limitations 
+	under the License. -->
+
+<project
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+	xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>io.druid.extensions</groupId>
+	<artifactId>druid-cloudfiles-extensions</artifactId>
+	<name>druid-cloudfiles-extensions</name>
+	<description>druid-cloudfiles-extensions</description>
+
+	<parent>
+		<groupId>io.druid</groupId>
+		<artifactId>druid</artifactId>
+		<version>0.8.2-SNAPSHOT</version>
+		<relativePath>../../pom.xml</relativePath>
+	</parent>
+
+	<properties>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+		<jclouds.version>1.9.1</jclouds.version>
+	</properties>
+
+	<dependencies>
+		<dependency>
+			<groupId>io.druid</groupId>
+			<artifactId>druid-api</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>com.metamx</groupId>
+			<artifactId>emitter</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>com.google.inject</groupId>
+			<artifactId>guice</artifactId>
+			<version>3.0</version><!--$NO-MVN-MAN-VER$ -->
```

Comment:
I believe the reason it "works" is because you are explicitly declaring a dependency on guice 3.0, so guice-3.0.jar will be present in the classpath or loaded by isolated classloader (depending on which way you load extensions). Suppose there is a class X exists in guice 3.0 but not in 4.0, then there won't be a problem because classloader will eventually find that in ```guice-3.0.jar``` . However, if class X exists both in guice 3.0 and 4.0, then X in guice 4.0 will take precedence, which could be a potential problem if class X has backward-incompatible changes from 3.0 to 4.0.